addons/hr_timesheet/models/project.py

```diff
@@ -367,7 +367,7 @@ def _unlink_except_contains_entries(self):
         In this case, a warning message is displayed through a RedirectWarning
         and allows the user to see timesheets entries to unlink.
         """
-        tasks_with_timesheets = self.filtered(lambda t: t.timesheet_ids)
+        tasks_with_timesheets = self.filtered(lambda t: t.sudo().timesheet_ids)
```

Comment:
Since we just want to know if any tasks in `self` have at least one timesheet, you could perhaps do a `read_group` on the hr_timesheet model (`account.analytic.line` one) to get all task ids with at least one timesheet instead of prefetching all timesheets contained in all tasks in `self`. What do you think? 🙂 

Something like that:
```suggestion
        timesheet_read_group = self.env['account.analytic.line'].sudo().read_group(
            [('task_id', 'in', self.ids)],
            ['task_id'],
            ['task_id'],
        )
        task_ids_with_timesheets = [res['task_id'][0] for res in timesheet_read_group]
```
To forget to adapt the code below too. 🙂 

Normally, if I am not wrong, we will also improve the performance when there are many tasks in `self`. 🙂

Reply:
Yes, this'll improve the performance in multi-mode for sure 😸

Reply:
@xavierbol I edited it, pls check again